### PR TITLE
migrate @babel/preset-env to babel-preset-modules

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

File: build-system/compile/compile.js [hunk 1]
```diff
@@ -256,8 +256,8 @@ function compile(entryModuleFilenames, outputDir, outputFilename, options) {
       assume_function_wrapper: true,
       // Transpile from ES6 to ES5 if not running with `--esm`
       // otherwise transpilation is done by Babel
-      language_in: 'ECMASCRIPT6',
-      language_out: argv.esm ? 'NO_TRANSPILE' : 'ECMASCRIPT5',
+      language_in: argv.esm ? 'ECMASCRIPT_2017' : 'ECMASCRIPT6',
+      language_out: argv.esm ? 'ECMASCRIPT_2017' : 'ECMASCRIPT5',
```
Comment (kristoferbaxter) [inline build-system/compile/compile.js]: Is this change necessary? 

Why switch to `ECMASCRIPT_2017` which could perform some transpilation when `NO_TRANSPILE` guarantees no additional transpilation from Closure Compiler?

File: build-system/compile/single-pass.js [hunk 2]
```diff
@@ -105,14 +105,16 @@ exports.getFlags = function(config) {
...
     // By default closure puts all of the public exports on the global, but
     // because of the wrapper modules (to mitigate async loading of scripts)
     // that we add to the js binaries this prevents other js binaries from
     // accessing the symbol, we remedy this by attaching all public exports
     // to `_` and everything imported across modules is is accessed through `_`.
     rename_prefix_namespace: '_',
-    language_out: config.language_out || 'ES5',
+    language_in: config.esm ? 'ECMASCRIPT_2017' : 'ECMASCRIPT6',
+    language_out: config.esm
+      ? 'ECMASCRIPT_2017'
```
Comment (kristoferbaxter) [inline build-system/compile/single-pass.js]: Same question as above here: Should we use `NO_TRANSPILE` instead?

File: build-system/typescript.js [hunk 3]
```diff
@@ -36,8 +36,8 @@ exports.transpileTs = function(srcDir, srcFilename) {
   const tsEntry = path.join(srcDir, srcFilename).replace(/\.js$/, '.ts');
   const tsConfig = ts.convertCompilerOptionsFromJson(
     {
-      'module': 'ES6',
-      'target': 'ES6',
+      'module': 'esnext',
```
Comment (kristoferbaxter) [inline build-system/typescript.js]: 👍

File: build-system/compile/compile.js (hunk 1 above)
Comment (erwinmombay) [reply build-system/compile/compile.js]: no i dont believe this is necessary. was just testing the outputs out. will revert back to NO_TRANSPILE

Comment (erwinmombay) [reply build-system/compile/compile.js]: `language_out` should be now `NO_TRANSPILE` when `esm` flag is used

File: build-system/compile/single-pass.js (hunk 2 above)
Comment (erwinmombay) [reply build-system/compile/single-pass.js]: `language_out` should be now `NO_TRANSPILE` when `esm` flag is used

File: build-system/compile/compile.js [hunk 4]
```diff
@@ -256,7 +256,7 @@ function compile(entryModuleFilenames, outputDir, outputFilename, options) {
       assume_function_wrapper: true,
       // Transpile from ES6 to ES5 if not running with `--esm`
       // otherwise transpilation is done by Babel
```
Comment (rsimha) [inline build-system/compile/compile.js]: Does this comment need to be updated?

File: package.json [hunk 5]
```diff
@@ -20,6 +20,7 @@
   "dependencies": {
     "@ampproject/animations": "0.2.0",
     "@ampproject/worker-dom": "0.19.0",
+    "babel-preset-modules": "0.0.2",
```
Comment (rsimha) [inline package.json]: Should `@babel/preset-env` have been removed from `package.json`?
